### PR TITLE
CLN: Clean test_arithmetic.py

### PR DESCRIPTION
**Pull request by dsaxton**
General cleaning (i.e., replacing for loops) trying to make these easier to follow / debug

### CODE REVIEW

File: pandas/tests/series/test_arithmetic.py [hunk 1]
```diff
@@ -553,68 +551,83 @@ def test_comparison_tuples(self):
         expected = Series([True, False])
         tm.assert_series_equal(result, expected)
 
-    def test_comparison_operators_with_nas(self):
+    @pytest.mark.parametrize("op", ["lt", "le", "gt", "ge", "eq", "ne"])
```
Comment (mroeschke) [inline pandas/tests/series/test_arithmetic.py:554]: I think these exist as dunder strings in the `all_compare_operators` fixture

File: pandas/tests/series/test_arithmetic.py [hunk 2]
```diff
@@ -271,7 +271,6 @@ def test_comparison_flex_basic(self):
         tm.assert_series_equal(left.gt(right), left > right)
         tm.assert_series_equal(left.ge(right), left >= right)
 
-        # axis
         for axis in [0, None, "index"]:
```
Comment (jreback) [inline pandas/tests/series/test_arithmetic.py:274]: could parameterize here (might have to split the test)

File: pandas/tests/series/test_arithmetic.py [hunk 3]
```diff
@@ -280,7 +279,6 @@ def test_comparison_flex_basic(self):
             tm.assert_series_equal(left.gt(right, axis=axis), left > right)
             tm.assert_series_equal(left.ge(right, axis=axis), left >= right)
 
-        #
         msg = "No axis named 1 for object type"
```
Comment (jreback) [inline pandas/tests/series/test_arithmetic.py:282]: could use the fixture

**Review by jreback — APPROVED** [review]
thanks @dsaxton some comments, but happy to do in a followon.

**Pull request merged**